glib/src/property.rs

```diff
@@ -169,6 +179,41 @@ impl<T> PropertySet for once_cell::unsync::OnceCell<T> {
     }
 }
 
+impl<T: IsA<Object>> PropertyGet for WeakRef<T> {
+    type Value = T;
+
+    fn get<R, F: Fn(&Self::Value) -> R>(&self, f: F) -> R {
+        match self.upgrade() {
+            Some(value) => f(&value),
+            None => panic!("failed to upgrade weak reference"),
```

Comment:
Is that really the behaviour we want? I would've expected it to simply return `None`/`NULL` in the property instead of panicking.

Reply:
Changed it to use `Option<T>`